etc/notes/CHANGES_5.0.0.md

```diff
@@ -18,6 +18,84 @@ The following is a detailed collection of the changes in the major v5 release of
 
 ### Optional callback support migrated to `mongodb-legacy`
 
+Node v5 drops support for callbacks in favor of a pure async-await API.  Below are some strategies for
+callback users to adopt driver v5 in recommendation order.
+
+The Node driver team understands that a callback to promise migration can be a non-trivial refactor.  To help inform your migration strategy, we've outlined three different approaches below.
+
+#### Migrate to promise based api (recommended!)
+
+The Node team strongly encourages anyone who is able to migrate from callbacks to promises to do so.  Adopting the
+regular driver API will streamline the adoption of future driver updates, as well as provide better Typescript support
+than the other options outlined in this document.
+
+The promise-based API is identical to the callback API except:
+
+- no callback is accepted as the last argument
+- a promise is always returned
+
+For example, with a findOne query:
+
+```typescript
+// callback-based API
+collection.findOne({ name: 'john snow'}, (error, result) => {
+  if (error) {
+    doSomethingWithError(error);
+    return;
+  }
+
+  doSomethingWithResult(result);
+})
+
+// promise-based API
+collection.findOne({ name: 'john snow'}).then(
+  (result) => doSomethingWithResult(result),
+  (error) => doSomethingWithError(error)
+);
+
+// promise-based API with async/await
+try {
+  const result = await collection.findOne({ name: 'john snow'});
+  doSomethingWithResult(result);
+} catch (error) {
+  doSomethingWithError(error);
+}
+```
+
+#### Use the promise-based API and `util.callbackify`
+
+If you only have a few callback instances where you are currently unable to adopt the promise API, we recommend using the promise API and Nodejs's `callbackify`
+utility to adapt the promise-based API to use callbacks.
+
+**Note** Manually converting a promise-based api to a callback-based API is error prone.  We strongly encourage the use of `callbackify`.
+
+There are two approaches to using `callbackify` with a driver api.
+
+First, the driver method can be callbackified.  If this approach is used, care must be taken to bind the
```

Comment:
I think we should make the anonymous function approach first, and potentially consider dropping the bind one. The anonymous one allows folks to continue omitting the options argument, when you use bind I think you need to provide options to make the callback the third argument, which I don't think is worth explaining.

Reply:
I'm fine removing the `bind` approach.  Although, it does preserve optional arguments.